#!/usr/bin/env python
PACKAGE = "vision_tasks"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("clipLimit", double_t, 0, "Histogram Clip Limit", 4, 0, 80)
gen.add("tileGridSize", int_t, 0, "CLAHE Grid Size", 3, 1, 7)
gen.add("whiteBalanceWeight", double_t, 0, "Weight of white balance", 0.5, 0.0, 1.0)
gen.add("h", double_t, 0, "Noise reduction strength", 4.0, 0.0, 40.0)
gen.add("hColor", double_t, 0, "Noise reduction strength (color)", 4.0, 0.0, 40.0)
gen.add("templateWindowSize", int_t, 0, "Noise reduction template size", 5, 0, 30)
gen.add("searchWindowSize", int_t, 0, "Noise reduction search window size", 11, 0, 30)

exit(gen.generate(PACKAGE, "vision_tasks", "blue_filter"))

